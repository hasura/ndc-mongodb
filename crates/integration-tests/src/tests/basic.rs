use crate::graphql_query;
use insta::assert_yaml_snapshot;
use serde_json::json;

#[tokio::test]
async fn runs_a_query() -> anyhow::Result<()> {
    assert_yaml_snapshot!(
        graphql_query(
            r#"
                query Movies {
                  movies(limit: 10, order_by: { id: Asc }) {
                    title
                    imdb {
                      rating
                      votes
                    }
                  }
                }
            "#
        )
        .run()
        .await?
    );
    Ok(())
}

#[tokio::test]
async fn filters_by_date() -> anyhow::Result<()> {
    assert_yaml_snapshot!(
        graphql_query(
            r#"
                query ($dateInput: Date) {
                  movies(
                    order_by: {id: Asc},
                    where: {released: {_gt: $dateInput}}
                  ) {
                    title
                    released
                  }
                }
            "#
        )
        .variables(json!({ "dateInput": "2016-03-01T00:00Z" }))
        .run()
        .await?
    );
    Ok(())
}

#[tokio::test]
async fn selects_array_within_array() -> anyhow::Result<()> {
    assert_yaml_snapshot!(
        graphql_query(
            r#"
            query {
              artistsWithAlbumsAndTracks(limit: 1, order_by: {id: Asc}) {
                name
                albums {
                  title
                  tracks {
                    name
                  }
                }
              }
            }
            "#
        )
        .run()
        .await?
    );
    Ok(())
}

#[tokio::test]
async fn selects_field_names_that_require_escaping() -> anyhow::Result<()> {
    assert_yaml_snapshot!(
        graphql_query(
            r#"
            query {
              testCases_weirdFieldNames(limit: 1, order_by: { invalidName: Asc }) {
                invalidName
                invalidObjectName {
                  validName
                }
                validObjectName {
                  invalidNestedName
                }
              }
            }
            "#
        )
        .run()
        .await?
    );
    Ok(())
}

#[tokio::test]
async fn selects_nested_field_with_dollar_sign_in_name() -> anyhow::Result<()> {
    assert_yaml_snapshot!(
        graphql_query(
            r#"
            query {
              testCases_nestedFieldWithDollar(order_by: { configuration: Asc }) {
                configuration {
                  schema
                }
              }
            }
            "#
        )
        .run()
        .await?
    );
    Ok(())
}
