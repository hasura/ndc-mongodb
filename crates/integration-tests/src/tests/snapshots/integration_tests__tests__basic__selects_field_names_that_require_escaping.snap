---
source: crates/integration-tests/src/tests/basic.rs
expression: "graphql_query(r#\"\n            query {\n              weirdFieldNames(limit: 1, order_by: { invalidName: Asc }) {\n                invalidName\n                invalidObjectName {\n                  validName\n                }\n                validObjectName {\n                  invalidNestedName\n                }\n              }\n            }\n            \"#).run().await?"
---
data:
  weirdFieldNames:
    - invalidName: 1
      invalidObjectName:
        validName: 1
      validObjectName:
        invalidNestedName: 1
errors: ~
